Add testing section to Readme (#243)
neo4j/neo4j-go-driver: README.md

For a customised logging target, you can implement the above interface and pass an instance of that implementation to the `BoltLogger` field.

> ## Developing

Reviewer: How about using `CONTRIBUTING.md` for this instead?
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#about-contributing-guidelines

Reviewer: ```suggestion
## Development
```

Author: All the other drivers have building and development instructions in the Readme. Doing it for the go driver makes it more consistent with the other repos.

> This section describe instructions to be used during the development phase of the driver. 

Reviewer: ```suggestion
This section describes instructions to use during the development phase of the driver. 
```

### Testing

Tests **require** the latest [Testkit 4.3](https://github.com/neo4j-drivers/testkit/tree/4.3), Python3 and Docker.

> Testkit is needed to be cloned and configured to run against the Go Driver. Use the following steps to configure Testkit.

Reviewer: ```suggestion
Testkit needs to be cloned and configured to run against the Go Driver. Use the following steps to configure it.
```

1. Clone the Testkit repository

> ```

Reviewer: ```suggestion
```shell
```

git clone https://github.com/neo4j-drivers/testkit.git
```

2. Under the Testkit folder, install the requirements.

> ```

Reviewer: ```suggestion
```shell
```

pip3 install -r requirements.txt
```

> 3. Define some enviroment variables to configure Testkit

Reviewer: ```suggestion
3. Define the following required environment variables
```

> ```

Reviewer: ```suggestion
```shell
```

export TEST_DRIVER_NAME=go
export TEST_DRIVER_REPO=<path for the root folder of driver repository>
```

To run test against against some Neo4j version:

> ```

Reviewer: ```suggestion
```shell
```

python3 main.py
```

> More details about how to use Teskit could be found on [its repository](https://github.com/neo4j-drivers/testkit/tree/4.3)

Reviewer: ```suggestion
More details about how to use Teskit can be found on [its repository](https://github.com/neo4j-drivers/testkit/tree/4.3)
```
